python/proton/_utils.py

```diff
@@ -440,11 +440,16 @@ def close(self):
                 self.conn.close()
                 self.wait(lambda: not (self.conn.state & Endpoint.REMOTE_ACTIVE),
                           msg="Closing connection")
+                if self.conn.transport:
+                    # Close tail to force transport cleanup without waiting/hanging for peer close frame.
+                    self.conn.transport.close_tail()
         finally:
             self.conn.free()
             # Nothing left to block on.  Allow reactor to clean up.
             self.run()
-            self.conn = None
+            if self.conn:
+                self.conn.handler = None  # break cyclical reference
+                self.conn = None
             self.container.global_handler = None  # break circular ref: container to cadapter.on_error
```

Comment:
This is probably no longer necessary (according to the comment) as we aren't using cadapter.on_error any more.